### PR TITLE
Handle 2-pattern-direction-with-same-headsign edge case

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**
There was a strange edge case where there could be 2 patterns with the same head sign. we fixed this by appending the stop info to the second head sign but this looked really weird. this PR fixes this by addressing this one specific edge case

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by amy-corson-ibigroup — APPROVED** [review]

File: lib/components/viewers/route-details.tsx [hunk 1]
```diff
@@ -108,9 +108,27 @@ class RouteDetails extends Component<Props> {
...
+            amended[0].headsign = intl.formatMessage(
+              { id: 'components.RouteDetails.headsignTo' },
+              { ...amended[0] }
+            )
+          }
         }
 
-        amended.push(cur)
+        if (alreadyExistingIndex >= 0) {
+          // Only replace if new pattern has greater geometry
```
Comment (binh-dam-ibigroup) [inline lib/components/viewers/route-details.tsx:123]: I recall that if we fuse patterns, at one point there was an issue navigating from the route viewer to the pattern viewer. Specifically, the pattern viewer would not show a pattern by default. Can you check that the issue is not reintroduced?

Comment (daniel-heppner-ibigroup) [reply lib/components/viewers/route-details.tsx:123]: Are you referring to the bug that I fixed here? https://github.com/opentripplanner/otp-react-redux/pull/1137

Comment (miles-grant-ibigroup) [reply lib/components/viewers/route-details.tsx:123]: Yes I can confirm that this bug isn't happening anymore with this branch

File: lib/components/viewers/route-details.tsx [hunk 2]
```diff
@@ -108,9 +108,27 @@ class RouteDetails extends Component<Props> {
...
+          if (amended.length === 1 && Object.entries(patterns).length === 2) {
+            amended[0].headsign = intl.formatMessage(
+              { id: 'components.RouteDetails.headsignTo' },
+              { ...amended[0] }
+            )
+          }
         }
 
-        amended.push(cur)
+        if (alreadyExistingIndex >= 0) {
```
Comment (binh-dam-ibigroup) [inline lib/components/viewers/route-details.tsx:122]: Add a comment regarding showing only one pattern if there were two patterns originally and the two patterns have the same text.

**Review by binh-dam-ibigroup — APPROVED** [review]
Add the comment as indicated.

**Pull request merged**